cpp/src/lance/arrow/file_lance.cc

```diff
@@ -135,4 +139,35 @@ bool IsLanceFragmentScanOptions(const ::arrow::dataset::FragmentScanOptions& fso
   return fso.type_name() == kLanceFormatTypeName;
 }
 
+LanceFragment::LanceFragment(std::shared_ptr<::arrow::fs::FileSystem> fs,
+                             std::string data_dir,
+                             std::shared_ptr<lance::format::DataFragment> fragment,
+                             const format::Schema& schema)
+    : fs_(std::move(fs)),
+      data_uri_(std::move(data_dir)),
+      fragment_(std::move(fragment)),
+      schema_(schema) {}
+
+::arrow::Result<::arrow::RecordBatchGenerator> LanceFragment::ScanBatchesAsync(
+    const std::shared_ptr<::arrow::dataset::ScanOptions>& options) {
+  // Only support one file for now.
+  assert(fragment_->data_files().size() == 1);
```

Comment:
So for now it won't support our current coco dataset with like 17 files then?

Reply:
In coco, each file is one `Fragment`.  So to support easy add / remove columns without re-write the dataset, a `Fragment` is a logic chunk of table (i.e., N rows), while it is flexible for `lance` to combine several files together to reconstruct these rows. 

For example, table rows [1:100] can be all in the same `Fragment`, while `Fragment.files[0]` is the first 5 columns, and `Fragment.files[1]` can be the newly added column (i.e,. `predict_of_model_A`), and we can still add one more file with one column the the `Fragment` for `predict_of_model_B` later.

Reply:
ic, so each version can have multiple fragments but _for now_ a fragment means 1 file?

Reply:
Yes, that's right.